### PR TITLE
Trigger checkpointing on SIGTERM

### PR DESCRIPTION
**Pull request by gwenzek**
Since #1603 we don't use the chaining of SIGTERM / SIGUSR1 to detect between preemption and timeout.
Some Slurm cluster aren't configured with `preempt_send_user_signal` and don't receive the `SIGUSR1`, so we should checkpoint as soon as we receive SIGTERM.

### CODE REVIEW

File: submitit/core/job_environment.py [hunk 1]
```diff
@@ -130,8 +130,7 @@ def _handle_signals(self, paths: JobPaths, submission: DelayedSubmission) -> Non
         @plugin-dev: Should be adapted to the signals used in this cluster.
         """
         handler = SignalHandler(self, paths, submission)
-        signal.signal(signal.SIGUSR1, handler.checkpoint_and_try_requeue)
-        signal.signal(signal.SIGTERM, handler.bypass)
+        signal.signal(signal.SIGTERM, handler.checkpoint_and_try_requeue)
```
Comment (jrapin) [inline submitit/core/job_environment.py:133]: Can you add a signal handler for sigcont and sigusr1 (ust logging that a signal was received, nothing more? This can be helpful for future updates

File: integration/preemption.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        slurm_additional_parameters={},
+    )
+    if node:
+        ex.update_parameters(slurm_additional_parameters={"nodelist": node})
+
+    return ex.submit(clock, partition, timeout_min)
+
+
+def wait_job_is_running(job: Job) -> None:
+    while job.state != "RUNNING":
```
Comment (jrapin) [inline integration/preemption.py:60]: may be safer check if it is still PENDING or UNKNOWN (can't remember if that's exactly it). If it fails for some unknown reason, this could wait forever

File: integration/preemption.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        log.info(f"{job} is not RUNNING")
+        time.sleep(60)
+
+
+def sacct(job: Job, field: str):
+    sacct = subprocess.check_output(
+        ["sacct", "-j", job.job_id, "-o", f"JobID,{field}", "--parsable2"], text=True,
+    )
+    node = next((l.split("|")[-1] for l in sacct.splitlines() if l.startswith(job.job_id + "|")), None)
```
Comment (jrapin) [inline integration/preemption.py:69]: is there a way to get this info directly through `Job`? may be useful eventually

File: submitit/slurm/_sbatch_test_record.txt [hunk 4]
```diff
@@ -10,7 +10,7 @@
 #SBATCH --open-mode=append
 #SBATCH --output=/tmp/%j_0_log.out
 #SBATCH --partition=learnfair
-#SBATCH --signal=USR1@90
+#SBATCH --signal=TERM@90
```
Comment (jrapin) [inline submitit/slurm/_sbatch_test_record.txt:13]: Is this change necessary now? :s don't we get it anyway?

Comment (gwenzek) [reply submitit/slurm/_sbatch_test_record.txt:13]: we always get it, but we want to be able to control the delay.

File: submitit/core/job_environment.py (hunk 1 above)
Comment (gwenzek) [reply submitit/core/job_environment.py:133]: done

**Review by jrapin — APPROVED** [review]

File: integration/preemption.py (hunk 3 above)
Comment (gwenzek) [reply integration/preemption.py:69]: this only work for SlurmJob, so wasn't sure if we should add it or not to the API.

Comment (gwenzek) [reply integration/preemption.py]: after discussion, NodeList will be collected by the SlurmInfoWatcher and is readable through job.get_info()
